### ISSUE TITLE
intel igpu support (needs developer help)

### ISSUE THREAD

**Issue by xkenx1995** [stevezau/plex_generate_vid_previews#50]
i know amd support was added when you dont have one. is it possible to implement intel's igpu?

**Comment by stevezau** [comment]
@xkenx1995 if you know how to detect the intel igpu and what settings ffmpeg requires then yes i could add it?

**Comment by Mickman1** [comment]
Any word on this? I'm interested in running on an N100 or an Arc card.

**Comment by confuzedplayer** [comment]
wish i could provide information for this but I'm not very familiar with how all this works but i am interested in having this for my arc card as well.

**Comment by stevezau** [comment]
I'm travelling at the moment, back next week.. If you can wait I'll be able to help.

**Comment by thehamisreal** [comment]
Hey @stevezau i've just picked up a Intel Arc a310 to replace a Nvidia t400 that i was using for thumbnail generation via docker. I'd be happy to help test the above out when you're back on board.

**Comment by stevezau** [comment]
@thehamisreal it is on my list, ill get it to when i get some time. I did start on it but it is a bit more complicated than i had expected so just required more focus.

**Comment by jeremysherriff** [comment]
Should be something like `ffmpeg -hwaccel vaapi -hwaccel_device /dev/dri/renderD128` 
Although `qsv` may be better than `vaapi` - my experiences on this topic are minimal but I have been successfully using vaapi when running TDarr (https://github.com/HaveAGitGat/Tdarr) for both decode and encode phases for most media in my library.

Some potential info for checking ffmpeg capabilities and profiles:
- https://www.reddit.com/r/ffmpeg/comments/14fcmh5/getting_ffmpeg_to_use_vaapi_acceleration_when/
- https://emby.media/support/articles/Hardware-Acceleration-on-Linux.html

I'm running a docker version of this project, can help with testing.

**Comment by stevezau** [comment]
Hi @thehamisreal @xkenx1995  @jeremysherriff are you still waiting on this?

I haven't had time to focus on it and prob won't for a while.. If you actually need it I will prioritise it earlier but if not I might wait for someone who ping saying they need it. LMK thanks

**Comment by thehamisreal** [comment]
@stevezau yeah I'm keen and happy to get testing if that can help? It'll save me running both an arc and NVIDIA card ongoing.

**Comment by jeremysherriff** [comment]
I'm easy, I have processed my back catalog and am using native Plex schedule/processing to process new stuff. I am happy to test if needed.

**Comment by mikemulhearn** [comment]
+1 for intel iGPU here

**Comment by stevezau** [comment]
Unfortunately, I haven't had time to spend on this and probably won't for a long time. If anyone else has cycles please feel free to open a PR.

**Comment by chrismseeger** [comment]
[https://github.com/stevezau/plex_generate_vid_previews/pull/66](url)
